test/solidity/gatherSources.test.ts

```diff
@@ -0,0 +1,101 @@
+import { expect } from "chai";
+import MockFs from "mock-fs";
+import { gatherSources, ResolverEngine, SolidityImportResolver, ImportFile } from "../../src";
+
+describe("gatherSources function", function() {
+    let resolver: ResolverEngine<ImportFile> = SolidityImportResolver();
+
+     afterEach(function() {
+        MockFs.restore();
+    });
+    
+    it("gathers files included by given file", async function() {
+        MockFs({
+            "mainfile.sol": 'blahblah;\nimport "./otherfile.sol";\nimport "./somethingelse.sol";\nrestoffileblahblah',
```

Comment:
move the filenames and file sources into constants (array of pairs?) in both expect and mockfs